src/messaging/ExchangeContext.h

```diff
@@ -0,0 +1,318 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the classes corresponding to CHIP Exchange Context.
+ *
+ */
+
+#pragma once
+
+#include <support/BitFlags.h>
+#include <support/DLLUtil.h>
+#include <system/SystemTimer.h>
+#include <transport/SecureSessionMgr.h>
+
+#define EXCHANGE_CONTEXT_ID(x) ((x) + 1)
+
+using namespace chip::Transport;
```

Comment:
I would suggest dropping `using` statements from headers as it causes headaches (e.g. we have namespaces `chip::Platform` and `chip::System::Platform` and that created ambiguities on 'Platform' whenever both chip and chip::System using statements were used).

Reply:
I dropped the using statement in this PR. But I don't think we have convention not using "using namespace", this helps a lot to make the code simple and clear to read. Currently, we use this pattern throughout our code base, if it causes name confusion, it indicates we have naming issue.